smolyay/basis.py

```diff
@@ -72,6 +80,7 @@ class ChebyshevFirstKind(BasisFunction):
     def __init__(self,n):
         super().__init__()
         self._n = n
+        self._derivative = ChebyshevSecondKind(self._n-1)
```

Comment:
I know I wrote `self._derivative` on Slack, but it might be better to choose a clearer variable name here, since this is not actually the derivative. Maybe:
```suggestion
        self._derivative_polynomial = ChebyshevSecondKind(self._n-1)
```
?